tests/shopify_python/test_git_utils.py

```diff
@@ -0,0 +1,172 @@
+import os
+import py  # pylint: disable=unused-import
+import pytest
+from git import repo
+from shopify_python import git_utils
+
+@pytest.fixture
+def python_file(main_repo):
+    # type: (repo.Repo) -> str
+    file_text = "import os\n" \
+                "def foo():\n" \
+                "    return 4\n"
+    file_path = os.path.join(main_repo.working_dir, 'program.py')
+
+    with open(file_path, 'w') as writing_file:
+        writing_file.write(file_text)
+    return file_path
+
+@pytest.fixture
+def python_script(main_repo):
+    # type: (repo.Repo) -> str
+    file_text = "#!/usr/bin/env python3\n" \
+                "import os\n" \
+                "def bar():\n" \
+                "    return 6\n"
+    file_path = os.path.join(main_repo.working_dir, 'program')
+
+    with open(file_path, 'w') as writing_file:
+        writing_file.write(file_text)
+    return file_path
+
+@pytest.fixture
+def non_python_file(main_repo):
+    # type: (repo.Repo) -> str
+    file_text = "[MASTER]\n" \
+                "\n" \
+                "# Minimum lines number of a similarity.\n" \
+                "min-similarity-lines=4\n"
+    file_path = os.path.join(main_repo.working_dir, 'non_program')
+    with open(file_path, 'w') as writing_file:
+        writing_file.write(file_text)
+    return file_path
+
+@pytest.fixture
+def remote_repo(tmpdir):
+    # type: ('py.path.LocalPath') -> repo.Repo
+    return repo.Repo.init(str(tmpdir.join('remote')), bare=True)
+
+@pytest.fixture
+def main_repo(tmpdir, remote_repo):
+    # type: ('py.path.LocalPath', repo.Repo) -> repo.Repo
+    to_dir = str(tmpdir.join('main'))
+    new_repo = repo.Repo.clone_from(remote_repo.git_dir, to_dir)
+
+    file_name = os.path.join(new_repo.working_dir, 'new_file')
+    open(str(file_name), 'wb').close()
+    new_repo.index.add([str(file_name)])
+    new_repo.index.commit("initial commit")
+    new_repo.remote('origin').push('master')
+
+    return new_repo
+
+def test_detects_changed_python_files(main_repo, python_file, python_script):
+    # type: (repo.Repo, str, str) -> None
+
+    main_repo.create_head('foo').checkout()
+    main_repo.index.add([python_file, python_script])
+    main_repo.index.commit("adding python files")
+
+    changed_files = git_utils.changed_python_files_in_tree(main_repo.working_dir)
+    assert sorted(changed_files) == ['program', 'program.py']
```

Comment:
Maybe you could get `program` and `program.py` from the test fixtures? E.g. a function that performs `os.path.split(python_file)[-1]`?